phoenix-core/src/main/java/org/apache/hadoop/hbase/regionserver/IndexHalfStoreFileReaderGenerator.java

```diff
@@ -80,6 +81,9 @@ public Reader preStoreFileReaderOpen(ObserverContext<RegionCoprocessorEnvironmen
         HRegionInfo childRegion = region.getRegionInfo();
         byte[] splitKey = null;
         if (reader == null && r != null) {
+            if(!p.toString().contains(QueryConstants.LOCAL_INDEX_COLUMN_FAMILY_PREFIX)) {
+                return super.preStoreFileReaderOpen(ctx, fs, p, in, size, cacheConf, r, reader);
+            }
```

Comment:
Does the usage of this IndexHalfStoreFileReader go through limited private HBase APIs or are we still using private APIs here?